validation.go

```diff
@@ -377,80 +380,46 @@ func validateField(d *datadictionary.DataDictionary,
 		return nil
 	}
 
+	// Validate against allowed enumerations
 	allowedValues := d.FieldTypeByTag[int(field.tag)].Enums
 	if len(allowedValues) != 0 {
 		if _, validValue := allowedValues[string(field.value)]; !validValue {
 			return ValueIsIncorrect(field.tag)
 		}
 	}
+	// Define field type mappings for easy lookup
+	fieldTypeMapping := map[string]FieldValue{
```

Comment:
This instantiation performs a lot of unnecessary allocations which will negatively impact performance